// ToiChat Buffer Protocol 
// Messages
//
// Created on: 02/01/2016
// Author: Toi-Group
//
package ToiChatProtocol;

message ToiChatMessage {
    oneof messageType {
        DnsMessage dnsMessage = 1;
        ChatMessage chatMessage = 2;
    }
}

message Identifier {
    // Gather message about client sending a new message
    //
    required string clientName = 1; // Hostname/CallSign
    required string clientId = 2; // IPaddress
    required string dateAdded = 3;
    optional string description = 4 [default = "NONE"]; // Misc Info.
}

message DnsMessage {
    required Identifier id = 1;
    required string command = 2;
    optional string status = 3;
    repeated Identifier clients = 4;
}

message ChatMessage {
    required Identifier id = 1;
    repeated string recipients = 2;
    required string textMessage = 3;
}

message AckMessage{
    required Identifier id = 1;
    required bool status = 2;
}